File: alumnos/lizcontreras/tareas/tarea03_ufo/estadistica_ufo.r
#!/usr/bin/env Rscript

f <- file("stdin")

open(f)

dat <- read.table(f)
n <- nrow(dat)
suma <- sum(dat$V1)
media <- round(mean(dat$V1), 0)
minimo <- min(dat$V1)
maximo <- max(dat$V1)
mediana <- median(dat$V1)
desviacion <- round(sd(dat$V1), 0)
varianza <- round(var(dat$V1), 0)

print(paste("N = ", n, sep = " "))
print(paste("Suma = ", suma, sep = " "))
print(paste("Media = " , media, sep = " "))
print(paste("Maximo = ", maximo, sep = " "))
print(paste("Minimo = ", minimo, sep = " "))
print(paste("Mediana = ", mediana, sep = " "))
print(paste("Desviacion estandar = ", desviacion, sep = " "))
print(paste("Varianza = ", varianza, sep = " "))

close(f)

File: alumnos/lizcontreras/tareas/tarea03_ufo/estadistica_ufo.py
#!/usr/bin/env python

import re
import sys
import math
from sys import stdin

dat = []
datdat = []
media = 0
maximo = 0
minimo = 0
mediana = 0
varianza = 0
desviacion = 0

while True:
    x = int(sys.stdin.readline())
    dat.append(x)
    datdat.append(x*x)
    if not x:
       break

n = len(dat)
media = int(sum(dat)/n)
maximo = max(dat)
minimo = min(dat)
varianza = int((sum(datdat)/n) - (media*media))
desviacion = int(math.sqrt(varianza))
# Mediana
dat_sort = sorted(dat)
mitad = n / 2
if n % 2 == 0:
   mediana = (dat_sort[mitad + 1] + dat_sort[mitad + 2]) / 2
else:
   mediana = dat_sort[mitad + 1]

print "suma", sum(dat)
print "n=", n
print "Media = ", media
print "Maximo = ", maximo
print "Minimo = ", minimo
print "Mediana = ", mediana
print "Desviacion estandar = ", desviacion
print "Varianza = ", varianza
